### ISSUE TITLE
Investigate strange 404 on certain scratches

### ISSUE THREAD

**Issue by ethteck** [decompme/decomp.me#473]
It appears as if something deep in the subprocess call is causing the error, which makes this kind of difficult to debug.

https://decomp.me/scratch/F7CKF

```
 File "/var/www/decomp.me/repo/backend/coreapp/views/scratch.py", line 361, in compile
    compilation = compile_scratch(scratch)
  File "/var/www/decomp.me/repo/backend/coreapp/views/scratch.py", line 60, in compile_scratch
    return CompilerWrapper.compile_code(
  File "/var/www/decomp.me/repo/backend/coreapp/compiler_wrapper.py", line 135, in compile_code
    compile_proc = sandbox.run_subprocess(
  File "/var/www/decomp.me/repo/backend/coreapp/sandbox.py", line 119, in run_subprocess
    return subprocess.run(
  File "/usr/lib/python3.9/subprocess.py", line 505, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.9/subprocess.py", line 1741, in _execute_child
    env_list.append(k + b'=' + os.fsencode(v))
  File "/usr/lib/python3.9/os.py", line 810, in fsencode
    filename = fspath(filename)  # Does type-checking of `filename`.
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

**Comment by ethteck** [comment]
https://decomp.me/scratch/F7CKF/export to download the data. It might be worth trying to reproduce this locally

**Comment by ethteck** [comment]
fixed in [90b7443](https://github.com/decompme/decomp.me/commit/90b7443b5f513bb4dad3a7fef85446c769e64eef) and the commit before it. there were some scratches for which diff_label was `None`, which isn't allowed
